collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -595,12 +598,23 @@ protected void onActivityResult(int requestCode, int resultCode,
             return;
         }
 
-        switch (requestCode) {
-            case BARCODE_CAPTURE:
+        // For handling results returned by the Zxing Barcode scanning library
+        IntentResult barcodeScannerResult = IntentIntegrator.parseActivityResult(requestCode, resultCode, intent);
+        if (barcodeScannerResult != null) {
+            if (barcodeScannerResult.getContents() == null) {
+                // request was canceled...
+                Timber.i("QR code scanning cancelled");
+            } else {
                 String sb = intent.getStringExtra("SCAN_RESULT");
```

Comment:
I am not sure if the later one will work as I used the way it was mentioned in the zxing library documentation. But if it's working then I guess it doesn't matter which one is used.